azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/data/CallHistoryRepository.kt

```diff
@@ -0,0 +1,100 @@
+                // Using a new db instance instead of caching one as we do not have a
+                // reliable event when to dispose it.
+                DbHelper(context).writableDatabase
+                    .use { db ->
+                        val values = ContentValues().apply {
+                            put(CallHistoryContract.COLUMN_NAME_CALL_ID, callId)
+                            put(CallHistoryContract.COLUMN_NAME_CALL_DATE, callDateTime.toInstant().toEpochMilli())
+                        }
+
+                        val result = db.insert(CallHistoryContract.TABLE_NAME, null, values)
+                        if (result == -1L) {
+                            logger.warning("Failed to save call history record.")
+                        }
+
+                        // Execute cleanup separately (not in one transaction) in case of it fails,
+                        // so it does not affect insert.
+                        cleanupOldRecords(db)
+                    }
+            }
+        }
+    }
+
+    override suspend fun getAll(): List<CallHistoryRecordData> {
+        return withContext(Dispatchers.IO) {
+            // Using a new db instance instead of caching one as we do not have a
+            // reliable event when to dispose it.
+            DbHelper(context).writableDatabase.use { db ->
+                cleanupOldRecords(db)
+                val items = mutableListOf<CallHistoryRecordData>()
+                db.rawQuery(
+                    "SELECT ${CallHistoryContract.COLUMN_NAME_ID}, " +
+                        "${CallHistoryContract.COLUMN_NAME_CALL_DATE}, " +
+                        "${CallHistoryContract.COLUMN_NAME_CALL_ID} " +
+                        "FROM ${CallHistoryContract.TABLE_NAME}",
+                    null
+                ).use {
+                    if (it.moveToFirst()) {
+                        val idColumnIndex = it.getColumnIndexOrThrow(CallHistoryContract.COLUMN_NAME_ID)
+                        val nameColumnIndex = it.getColumnIndexOrThrow(CallHistoryContract.COLUMN_NAME_CALL_ID)
+                        val dateColumnIndex = it.getColumnIndexOrThrow(CallHistoryContract.COLUMN_NAME_CALL_DATE)
+                        do {
+                            items.add(
+                                CallHistoryRecordData(
+                                    id = it.getInt(idColumnIndex),
+                                    callId = it.getString(nameColumnIndex),
+                                    callStartedOn = OffsetDateTime.ofInstant(
+                                        Instant.ofEpochMilli(it.getLong(dateColumnIndex)), ZoneId.systemDefault()
+                                    ),
+                                )
+                            )
+                        } while (it.moveToNext())
+                    }
+                }
+                return@withContext items
+            }
+        }
+    }
+
+    private fun cleanupOldRecords(db: SQLiteDatabase) {
```

Comment:
nit: I'd probably mark this as `suspend` to avoid accidentally calling this code outside of the IO dispatcher after some future refactor that shuffles things around.

Or simply inline this code and remove the function entirely.

Reply:
Was not able to make it suspend as it is in use from critical section